packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -1591,6 +1674,17 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     if (widget.selectionEnabled && pasteEnabled && widget.selectionControls?.canPaste(this) == true) {
       _clipboardStatus?.update();
     }
+
+    // If text direction has changed then our _whitespaceFormatter needs to be
+    // rerun.
+    if (_lastUsedWhitespaceFormatter?._baseDirection != _textDirection) {
+      _value = _whitespaceFormatter.format(_value);
+    }
+
+    widget.controller._setInputFormatters(
```

Comment:
the input formatters need to checked against the old input formatters associated with the controller, not the widget. Rebuilding from `Column(children: <Widget>[TextField('text1', inputFormatter1), Text('text2', inputFormatter2)])` to `Column(children: <Widget>[TextField('text2', inputFormatter2), Text('text1', inputFormatter1)])` should not reformat, without having to use global keys.